src/utils.ts

```diff
@@ -90,14 +90,51 @@ export function wrapCrashReporting(f) {
 }
 
 export function resolvePath(p: string, normalize: boolean = true) {
-    p = parseEnvVariables(p)
+    p = parseVSCodeVariables(p)
     p = p.replace(/^~/, os.homedir())
     p = normalize ? path.normalize(p) : p
     return p
 }
 
-function parseEnvVariables(p: string) {
-    return p.replace(/\${env:(.*?)}/g, (_, variable) => {
+/**
+ * Parse a subset of VSCode 'variables' in `p`, and return a string with the replacements.
+ *
+ * Specifically, we support:
+ *  - ${userHome}
+ *  - ${workspaceFolder}
+ *  - ${workspaceFolder:<FOLDER_NAME>}
+ *  - ${pathSeparator}
+ *  - ${env:<ENVIRONMENT_VARIABLE>}
+ *
+ * See https://code.visualstudio.com/docs/editor/variables-reference for definitions of the
+ * above.
+ *
+ * TODO: this replicates functionality present in core VSCode! The implementation of this
+ *  function be replaced once this issue is resolved:
+ *      https://github.com/microsoft/vscode/issues/46471
+ */
+function parseVSCodeVariables(p: string) {
+    p = p.replace(/\${userHome}/g, os.homedir())
+
+    const workspace_paths = vscode.workspace.workspaceFolders.map((folder) => {
+        return folder.uri.fsPath
+    })
+    p = p.replace(/\${workspaceFolder}/g, (_) => {
+        // We do not support multi-root workspaces here.
+        // In that case the user should use the ${workspaceFolder:<FOLDER_NAME>} syntax.
+        return workspace_paths.length === 1 ? workspace_paths[0] : null
```

Comment:
Elsewhere in the extension we're defaulting to the first workspace folder, which I think we should do here as well (could use `workspaceFolder1` for that to be explicit, I guess).

Reply:
OK, let's default to the first one here too then. (Introducing a new name seems like it could add more confusion... especially if it were only done here and not in the other places!)

Reply:
(Done)